src/charm.py

```diff
@@ -487,6 +488,14 @@ def _on_start(self, event: ops.charm.StartEvent):
         # available and the status was stuck on "Waiting for IP address". Adding this hook as a workaround.
         self._common_exit_hook()
 
+    def _on_install(self, _):
+        """Event handler for the install event during which we will update the K8s service"""
+        # At this point, it's debatable whether we should put the charm into a blocked state.
+        # My guess is right now, it's not essential, but as we start to wire these things more
+        # carefully, we should probably do so.
+        if not self._patch_k8s_service():
```

Comment:
Yeh so this could well end up us a small library or similar that could be shared across the O11y charms if it turns out to be a common enough requirement. This is just it's simplest implementation in many ways.

Reply:
Refactored. WDYT?